components/builder-depot/src/server.rs

```diff
@@ -2064,6 +2113,186 @@ pub fn download_latest_builder_key(req: &mut Request) -> IronResult<Response> {
     Ok(response)
 }
 
+pub fn create_origin_secret(req: &mut Request) -> IronResult<Response> {
+    let params = match validate_params(req, &["origin"]) {
+        Ok(p) => p,
+        Err(st) => return Ok(Response::with(st)),
+    };
+
+    let origin = &params["origin"];
+
+    let body = match req.get::<bodyparser::Struct<OriginSecretPayload>>() {
+        Ok(Some(body)) => {
+            if body.name.len() <= 0 {
+                return Ok(Response::with((
+                    status::UnprocessableEntity,
+                    "Missing value for field: `name`",
+                )));
+            }
+            if body.value.len() <= 0 {
+                return Ok(Response::with((
+                    status::UnprocessableEntity,
+                    "Missing value for field: `value`",
+                )));
+            }
```

Comment:
I'm so stoked you're validating this kind of stuff.  I wish we had a better mechanism for DRYing this kind of thing up in our codebase.  (And just to be clear, I realize that adding one is outside of the scope of this PR 😁 )